index.js

```diff
@@ -6,9 +6,12 @@ var supportsColor = require('supports-color');
 var defineProps = Object.defineProperties;
 var isSimpleWindowsTerm = process.platform === 'win32' && !/^xterm/i.test(process.env.TERM);
 
+var levelMapping = ['ansi', 'ansi', 'ansi256', 'ansi16m'];
+var skipModels = ['gray'];
+
 function Chalk(options) {
-	// detect mode if not set manually
-	this.enabled = !options || options.enabled === undefined ? supportsColor : options.enabled;
+	// detect level if not set manually
+	this.level = !options || options.level === undefined ? supportsColor.level : options.level;
```

Comment:
I personally think the code I posted is bloated and huge just to support a property that doesn't really make sense now that there are different levels of support instead of a boolean on/off value.

If we were to do this, then I agree; `.enabled` would have to reflect the `.level` given a getter/setter descriptor.